cob_twist_controller/cfg/TwistController.cfg

```diff
@@ -11,21 +11,27 @@ damping_method_enum = gen.enum([
                        gen.const("truncation",          int_t, 2, "truncation")],
                      "enum types for the damping_method")
 
-gen.add("damping_method", int_t, 0, "The damping method to use (edited via an enum)", 0, 0, 2, edit_method=damping_method_enum)
+constraints_enum = gen.enum([ 
+                       gen.const("None",           int_t, 0, "No constraints active"),
+                       gen.const("WLN",            int_t, 1, "Weighted-least-norm base, with identity as weighting matrix (equal to None)"),
+                       gen.const("WLN_JLA",        int_t, 2, "Weighted-least-norm joint-limit-avoidance")],
+                     "enum types for the constraints")
 
-gen.add("eps", double_t, 0, "Value for singular value threshold (used in TRUNCATION)",    0.001, 0,   1)
-gen.add("damping_factor", double_t, 0, "The constant damping_factor (used in CONSTANT)",  0.01, 0, 1)
-gen.add("lambda0", double_t, 0, "Value for maximum damping_factor (used in MANIPULABILITY)",  0.1, 0, 10)
-gen.add("wt", double_t, 0, "Value for manipulability threshold (used in MANIPULABILITY)",  0.005, 0, 1)
+gen.add("damping_method",    int_t,    0, "The damping method to use (edited via an enum)", 0, None, None, edit_method=damping_method_enum)
 
-gen.add("JLA_active",   bool_t,   0, "Joint Limit Avoidance active",  True)
-gen.add("enforce_limits",   bool_t,   0, "If 'True', normalize velocities in TwistController, else let RobotHW handle the limits",  True)
-gen.add("tolerance",    double_t, 0, "Tolerance around joint position limits where velocities are normalized [degrees]",  5, 0, 15)
+gen.add("eps",               double_t, 0, "Value for singular value threshold (used for truncation)",    0.001, 0,   1)
+gen.add("damping_factor",    double_t, 0, "The constant damping_factor (used in CONSTANT)",  0.01, 0, 1)
+gen.add("lambda0",           double_t, 0, "Value for maximum damping_factor (used in MANIPULABILITY)",  0.1, 0, 10)
+gen.add("wt",                double_t, 0, "Value for manipulability threshold (used in MANIPULABILITY)",  0.005, 0, 1)
 
-gen.add("base_compensation",   bool_t,   0, "Compensate base motion",  False)
-gen.add("base_active",   bool_t,   0, "Command base actively",  False)
-gen.add("base_ratio",    double_t, 0, "Value for base_ratio",  0.0, 0, 1)
-gen.add("reset_markers",   bool_t,   0, "Reset Markers",  False)
+gen.add("constraint",        int_t,    0, "The constraint to use (edited via an enum)", 2, None, None, edit_method=constraints_enum)
 
+gen.add("enforce_limits",    bool_t,   0, "If 'True', enforce joint position limits and normalize velocities in TwistController, else let RobotHW handle the limits",  True)
+gen.add("tolerance",         double_t, 0, "Tolerance around joint position limits where velocities are normalized [degrees]",  5, 0, 15)
```

Comment:
should we have a separate parameter for velocity_enforcing and joint_position_enforcing?

should we have a (boolean) parameter to toggle between "keep_direction"-scaling (i.e. scale_all_with_max_factor) and "individual"-scaling (i.e. scale_each_with_respective_factor)?